### ISSUE TITLE
3D point  is spelled `POINT Z` not `POINTZ`

### ISSUE THREAD

**Issue by michaelkirk** [georust/wkt#114]
I recently merged #91, but I think it contains a critical error. I do not use 3-d points, so it slipped past me. 

@ariesdevil - do you have interest in fixing this?

**Comment by kylebarron** [comment]
I thought the WKT spec allowed either `POINTZ` or `POINT Z`. I got a separate error from geozero when I passed in `POINTZ`

**Comment by kylebarron** [comment]
> Trying to find an actual spec for WKT is making me laugh out loud. I guess we should probably just "do whatever JTS does".

Yeah I was just searching for a spec and similarly couldn't find one. I suppose the [ISO one](https://www.iso.org/standard/60343.html) is the "official" one. But I think [this GEOS definition](https://libgeos.org/specifications/wkt/#wkt-bnf) is enough.

> I found this [loc.gov/preservation/digital/formats/fdd/fdd000548.shtml](https://loc.gov/preservation/digital/formats/fdd/fdd000548.shtml), which links to some hundred page documents, but they seem to be about WKT CRS, and doesn't include the string "POINTZ" or "POINT Z"

Yeah it is confusing that there's both WKT for geometries and for CRS. Sadly the original [simple features spec](https://www.ogc.org/standard/sfa/) only defines WKB not WKT.

**Comment by kylebarron** [comment]
At first glance it looks like _only_ point supports Z/M, and no other geometry types support it, so we should expand that support as well

**Comment by michaelkirk** [comment]
Ok, I'm working on making this change.

**Comment by kylebarron** [comment]
Sorry I should've given you a heads up that I was looking into this

**Comment by kylebarron** [comment]
FWIW while I was reading through the GEOS code [here](https://github.com/libgeos/geos/blob/8e525c825caf314d61a6279a426a59ded115681e/tests/unit/io/WKTReaderTest.cpp), I saw that they do [support both POINTZ and POINT Z](https://github.com/libgeos/geos/blob/8e525c825caf314d61a6279a426a59ded115681e/tests/unit/io/WKTReaderTest.cpp#L97-L102). It seems strange they support both in GEOS when Shapely only supports the latter. But it looks like it produces `POINT Z` by default https://github.com/libgeos/geos/blob/main/include/geos/io/WKTWriter.h#L165-L179

**Comment by kylebarron** [comment]
This can be closed from https://github.com/georust/wkt/pull/115
